draft-ietf-quic-transport.md

```diff
@@ -4037,36 +4082,38 @@ Note:
 data to a peer.
 
 ~~~
-       o
-       | Create Stream (Sending)
-       | Create Bidirectional Stream (Receiving)
-       v
-   +-------+
-   | Ready | Send RST_STREAM
-   |       |-----------------------.
-   +-------+                       |
-       |                           |
-       | Send STREAM /             |
-       |      STREAM_BLOCKED       |
-       v                           |
-   +-------+                       |
-   | Send  | Send RST_STREAM       |
-   |       |---------------------->|
-   +-------+                       |
-       |                           |
-       | Send STREAM + FIN         |
-       v                           v
-   +-------+                   +-------+
-   | Data  | Send RST_STREAM   | Reset |
-   | Sent  +------------------>| Sent  |
-   +-------+                   +-------+
-       |                           |
-       | Recv All ACKs             | Recv ACK
-       v                           v
-   +-------+                   +-------+
-   | Data  |                   | Reset |
-   | Recvd |                   | Recvd |
-   +-------+                   +-------+
+                            o
+                            | Create Stream (Sending)
+                            | Create Bidirectional Stream (Receiving)
+                            v
+                        +-------+
+                        | Ready | Send RST_STREAM
+                        |       |-----------------------.
+                        +-------+                       |
+                            |                           |
+                            | Send STREAM /             |
+                            |      STREAM_BLOCKED       |
+                            v                           |
+                        +-------+                       |
+Send STREAM /         .-|       |                       |
+     STREAM_BLOCKED / | | Send  | Send RST_STREAM       |
+Recv MAX_STREAM_DATA  '>|       |---------------------->|
+                        +-------+                       |
+                            |                           |
+                            | Send STREAM + FIN /       |
+                            | Recv STOP_SENDING         |
+                            v                           v
+                      .-+-------+                   +-------+
+    Recv STOP_SEDNING | | Data  | Send RST_STREAM   | Reset |
+                      '>| Sent  +------------------>| Sent  |
+                        +-------+                   +-------+
+                            |                           |
+                            | Recv All ACKs             | Recv ACK
+                            v                           v
+                        +-------+                   +-------+
+                        | Data  |                   | Reset |
+                        | Recvd |                   | Recvd |
+                        +-------+                   +-------+
```

Comment:
I don't think that the self-transitions are needed.  Those events simply don't cause changes to the current stream state, just like many other events that aren't shown.

Reply:
I added it because that fit with the text describing which messages can be send when.